.github/workflows/terraform-smoke.yml

```diff
@@ -75,6 +75,7 @@ jobs:
     - name: Set environment to configure provider for test
       run: |
         CONTROLLER=$(juju whoami --format yaml | yq .controller)
+        echo "JUJU_AGENT_VERSION=$(juju show-controller | yq .$CONTROLLER.details.agent-version |tr -d '"')" >> $GITHUB_ENV
```

Comment:
This can be improved. 
- Use `-r` flag of yq instead of trimming quotation marks
- use `.[]` instead of `.$CONTROLLER`. Since there's only 1 key they will be the same (however, on 2nd thought maybe worth keeping for consistency with other lines)

Reply:
I've added in -r and removed the `tr`.

I'll leave it as it is for consistency. I find on my machine I have to do `juju show-controller | yq --arg c "$CONTROLLER" '.[$c].details."agent-version"'` to get yq working consistently. It doesn't help that I often put dashes numbers and full stops in my controller names. I'll leave the `$CONTROLLER` as it is for now since it seems to work and its consistent with the others